target_chains/solana/programs/pyth-solana-receiver/src/lib.rs

```diff
@@ -0,0 +1,131 @@
+pub mod error;
+pub mod state;
+
+use {
+    crate::error::ReceiverError,
+    anchor_lang::prelude::*,
+    pythnet_sdk::{
+        accumulators::merkle::MerkleRoot,
+        hashers::keccak256_160::Keccak160,
+        messages::Message,
+        wire::{
+            from_slice,
+            v1::{
+                MerklePriceUpdate,
+                WormholeMessage,
+                WormholePayload,
+            },
+        },
+        ACCUMULATOR_EMITTER_ADDRESS,
+    },
+    serde::Deserialize,
+    sha3::Digest,
+    state::AnchorVaa,
+    std::io::Write,
+    wormhole_anchor_sdk::wormhole::{
+        SignatureSetData,
+        SEED_PREFIX_POSTED_VAA,
+    },
+};
+
+declare_id!("DvPfMBZJJwKgJsv2WJA8bFwUMn8nFd5Xpioc6foC3rse");
+pub const POST_VAA: u8 = 2;
+
+#[program]
+pub mod pyth_solana_receiver {
+    use super::*;
+
+    /// Verify the updates using the posted_vaa account
+    ///  * `vaa_hash` hash post of the post_vaa data to derive the address of the post_vaa account
+    ///  * `price_updates` Vec of bytes for the updates to verify and post on-chain
+    #[allow(unused_variables)]
+    pub fn post_updates(ctx: Context<PostUpdates>, price_updates: Vec<Vec<u8>>) -> Result<()> {
+        let vaa = &ctx.accounts.posted_vaa;
+        let signature_set = &ctx.accounts.signature_set;
+        require_keys_eq!(
+            vaa.meta.signature_set,
+            signature_set.key(),
+            ReceiverError::InvalidSignatureSet
+        );
+
+        let (expected_vaa_pubkey, _) = Pubkey::find_program_address(
+            &[SEED_PREFIX_POSTED_VAA, &signature_set.hash],
+            //TODO: expected program owner of the posted_vaa account should come from config account that can only be modified by governance
+            &wormhole_anchor_sdk::wormhole::program::id(),
+        );
+        require_keys_eq!(
+            vaa.key(),
+            expected_vaa_pubkey,
+            ReceiverError::InvalidVaaAccountKey
+        );
```

Comment:
What are the security requirements for verifying that the posted VAA is correctly signed? I don't understand why we need the signature_set here. How do other programs do this?

Reply:
this wormhole example seems like it's using the vaa_hash https://github.com/wormhole-foundation/wormhole-scaffolding/blob/main/solana/programs/01_hello_world/src/context.rs#L243

not sure if that's the current best practice or not. hard to tell from their docs.

Reply:
I still don't get passing the signature set signature set. 
A posted VAA is a VAA whose signatures have already been checked. 
People can pass whatever valid VAA to this program, if the proof doesn't match the VAA it will just fail.

Reply:
the signature set is just a different way to pass the `vaa_hash` since it also contains the hash of the data. this is just an extra check b/c like you said, if the proof doesn't match the vaa then it'll fail.

Reply:
I don't understand why you need this extra check, let's just keep it simple